### PR TITLE
Pulse edits and updates

### PR DESCRIPTION
**Pull request by taalexander**
### Summary
Updates the pulse defcalgrammar. Fixes ambiguities and other issues.

### CODE REVIEW

File: source/language/openpulse.rst [hunk 1]
```diff
@@ -235,7 +235,7 @@ samples provides optimization opportunities that wouldn't be available otherwise
 .. code-block:: javascript
 
    // arbitrary complex samples
-   waveform arb_waveform = [1+0*j, 0+1*j, 1/sqrt(2)+1/sqrt(2)*j];
+   waveform arb_waveform = [1+0im, 0+1im, 1/sqrt(2)+1/sqrt(2)im];
```
Comment (psivaraj) [inline source/language/openpulse.rst:238]: Should these use `*Im`?  Same deal with line 314. 

Also, should we add some statement about the complex number support still being in active development (see [here](https://github.com/Qiskit/openqasm/pull/301))?

File: source/language/pulses.rst [hunk 2]
```diff
@@ -108,10 +108,13 @@ Inline calibration blocks
 
 As calibration grammars may require the ability to insert top-level configuration information, perform program setup, or make inline calls
 to calibration-level instructions, OpenQASM supports the ability to declare a ``cal`` block. Within the ``cal`` block the
-semantics of the selected calibration grammar are valid. The ``cal`` block is of the same scope level as the enclosing block. The calibration
-grammar implementer may therefore choose to allow referencing or modifying values outside of the ``cal`` block but within the containing scope.
+semantics of the selected calibration grammar are valid. The calibration grammar implementer may therefore choose to
+allow referencing values (with its own syntax) from within the ``cal`` block that were declared within the containing parent scope.
+The ``cal`` block is of the same scope level as the enclosing block.
+The calibration grammar implementer may therefore choose to allow referencing or modifying values declared outside
+of the ``cal`` block but within the containing scope and its corresponding syntax.
```
Comment (psivaraj) [inline source/language/pulses.rst:115]: Looks like some text duplication here.

File: source/language/openpulse.rst (hunk 1 above)
Comment (taalexander) [reply source/language/openpulse.rst:238]: Added comment in 4f13bd9.

File: source/language/pulses.rst (hunk 2 above)
Comment (taalexander) [reply source/language/pulses.rst]: Took another stab at the language in c1a6862.

File: source/language/pulses.rst [hunk 3]
```diff
@@ -108,10 +108,12 @@ Inline calibration blocks
...
 to calibration-level instructions, OpenQASM supports the ability to declare a ``cal`` block. Within the ``cal`` block the
-semantics of the selected calibration grammar are valid. The ``cal`` block is of the same scope level as the enclosing block. The calibration
-grammar implementer may therefore choose to allow referencing or modifying values outside of the ``cal`` block but within the containing scope.
-Values declared within the ``cal`` block are only referenceable from other ``cal`` blocks or ``defcal`` declarations that may observe that scope as defined
-by the calibration grammar implementer. Values may not leak back to the enclosing blocks scope.
+semantics of the selected ``defcalgrammar`` are valid. The ``cal`` block is of the same scope level as the enclosing block. The
+calibration grammar may choose to allow capturing values (with chosen syntax) from within the ``cal``
+block that were declared within the containing parent scope.
+The calibration grammar may therefore choose to allow capturing values declared within the parent scope.
```
Comment (psivaraj) [inline source/language/pulses.rst:114]: Duplicate sentences? 

> The calibration grammar may choose to allow capturing values (with chosen syntax) from within the ``cal``
block that were declared within the containing parent scope. The calibration grammar may therefore choose to allow capturing values declared within the parent scope.

**Review by psivaraj — DISMISSED** [review]
Aside from that sentence structure, LGTM.

File: source/language/pulses.rst (hunk 3 above)
Comment (taalexander) [reply source/language/pulses.rst]: Fixed in 0cabc5f.

**Review by psivaraj — APPROVED** [review]

**Pull request merged**